components/script/dom/byteteeunderlyingsource.rs

```diff
@@ -0,0 +1,457 @@
+                    self.canceled_1.clone(),
+                    self.canceled_2.clone(),
+                    self.cancel_promise.clone(),
+                    self,
+                    global,
+                    can_gc,
+                );
+
+                let read_request = ReadRequest::ByteTee {
+                    byte_tee_read_request: Dom::from_ref(&byte_tee_read_request),
+                };
+
+                // Perform ! ReadableStreamDefaultReaderRead(reader, readRequest).
+                reader
+                    .get()
+                    .expect("Reader should be set.")
+                    .read(cx, &read_request, can_gc);
+            },
+        }
+    }
+
+    pub(crate) fn pull_with_byob_reader(
+        &self,
+        view: HeapBufferSource<ArrayBufferViewU8>,
+        for_branch2: bool,
+        cx: SafeJSContext,
+        global: &GlobalScope,
+        can_gc: CanGc,
+    ) {
+        let mut reader = self.reader.borrow_mut();
+        match &*reader {
+            ReaderType::BYOB(reader) => {
+                // Let byobBranch be branch2 if forBranch2 is true, and branch1 otherwise.
+                let byob_branch = if for_branch2 {
+                    self.branch_2.get().expect("Branch 2 should be set.")
+                } else {
+                    self.branch_1.get().expect("Branch 1 should be set.")
+                };
+
+                // let otherBranch be branch2 if forBranch2 is false, and branch1 otherwise.
+                let other_branch = if for_branch2 {
+                    self.branch_1.get().expect("Branch 1 should be set.")
+                } else {
+                    self.branch_2.get().expect("Branch 2 should be set.")
+                };
+
+                // Let readIntoRequest be a read-into request with the following items:
+                let byte_tee_read_into_request = ByteTeeReadIntoRequest::new(
+                    for_branch2,
+                    &byob_branch,
+                    &other_branch,
+                    self.read_again_for_branch_1.clone(),
+                    self.read_again_for_branch_2.clone(),
+                    self.reading.clone(),
+                    self.canceled_1.clone(),
+                    self.canceled_2.clone(),
+                    self.cancel_promise.clone(),
+                    self,
+                    global,
+                    can_gc,
+                );
+
+                let read_into_request = ReadIntoRequest::ByteTee {
+                    byte_tee_read_into_request: Dom::from_ref(&byte_tee_read_into_request),
+                };
+
+                // Perform ! ReadableStreamBYOBReaderRead(reader, view, 1, readIntoRequest).
+                reader.get().expect("Reader should be set.").read(
+                    cx,
+                    view,
+                    1,
+                    &read_into_request,
+                    can_gc,
+                );
+            },
+            ReaderType::Default(default_reader) => {
```

Comment:
Same here, you want to run the "If reader implements ReadableStreamDefaultReader" only for `ReaderType::Default`, but the other steps should run for both types.